src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineReceiver.php

```diff
@@ -46,6 +49,16 @@ public function get(): iterable
     {
         try {
             $doctrineEnvelope = $this->connection->get();
+            $this->retryingSafetyCounter = 0; // reset counter
+        } catch (RetryableException $exception) {
+            // Do nothing when RetryableException occurs less than "MAX_RETRIES"
+            // as it will likely be resolved on the next call to get()
+            // Problem with concurrent consumers and database deadlocks
+            if (++$this->retryingSafetyCounter >= self::MAX_RETRIES) {
+                throw new TransportException($exception->getMessage(), 0, $exception);
```

Comment:
It would be better to reset `retryingSafetyCounter` to 0 before throwing the exception. In case someone catched the error, the behavior should stay the same.

Reply:
updated